### DISCUSSION TITLE
Return calculated value on Select

### DISCUSSION THREAD

**Discussion by nogenem** [prisma/prisma#16808] [Q&A] [answered]
Hi guys, i would love to know if there is any way to replicate the following raw query in Prisma, without using `$queryRaw`:

```sql
SELECT
  p.name as productName,
  v.price as variantPrice,
  d.percent as discountPercent,
  CEIL((v.price / 100 * (1 - IFNULL(d.percent, 0) / 100 / 100)) * 100) discountedPrice
FROM Product p
LEFT JOIN Variant v 
  ON v.productId = p.id
LEFT JOIN Discount d
  ON d.id = p.discountId
ORDER BY discountedPrice;
```

In a simplified way, these are my models:

```prisma
model Product {
  id               String    @id @default(cuid())
  name             String    @db.Text
  variants         Variant[]
  category         Category  @relation(fields: [categoryId], references: [id])
  categoryId       String
  discount         Discount? @relation(fields: [discountId], references: [id])
  discountId       String?

  @@index([categoryId])
  @@index([discountId])
}

model Variant {
  id                 String         @id @default(cuid())
  price              Int
  product            Product        @relation(fields: [productId], references: [id])
  productId          String

  @@index([productId])
}

model Discount {
  id          String    @id @default(cuid())
  percent     Int
  products    Product[]
}
```

Variant.price and Discount.percent are stored as int, 7.50 = 750, hence the divisions and multiplications by 100. 😅

The final Query will be bigger than the one shown here and thats why i wanted to use Prisma' `findMany` function, but if there is no way to do the calculation, then i guess i can think a bit more about the final raw query. 🤣

**Comment by nurul3101** [comment, marked as answer]
Hey @nogenem 👋 
Based on the mathematical computation that you are doing with CEIL function, you would need to use Raw Query to achieve it, the CEIL function is not natively supported in Prisma.

**Reply by nogenem** [reply]
I ended up removing the CEIL, just leaving the rest of the calculation, but i also did the whole final query with a mix of Prisma.sql + Prisma.join + $queryRaw, after soferring a lot with it 😅

Anyways, thanks.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
